File: tools/repo_lint/tests/fixtures/perl/naming-violations.pl
#!/usr/bin/env perl
# Intentional naming convention violations for Perl.
#
# This file intentionally violates Perl naming conventions.
# It uses kebab-case instead of snake_case to test naming enforcement.

use strict;
use warnings;

# This file name itself (naming-violations.pl) violates snake_case
# Expected: naming_violations.pl
# Actual: naming-violations.pl (kebab-case - WRONG for Perl)

sub ThisSubShouldBeSnakeCase {
    # Subroutine names should be snake_case, not PascalCase
    return 1;
}

my $VariableInPascalCase = "wrong";  # Variables should be snake_case

sub normal_function {
    # This is correct snake_case naming
    return 1;
}

1;

File: tools/repo_lint/tests/fixtures/python/naming-violations.py
"""
Intentional naming convention violations for Python.

This file intentionally violates Python PEP 8 naming conventions.
It uses kebab-case instead of snake_case to test naming enforcement.
"""

from __future__ import annotations

# This file name itself (naming-violations.py) violates snake_case
# Expected: naming_violations.py
# Actual: naming-violations.py (kebab-case - WRONG for Python)


def ThisFunctionShouldBeSnakeCase():
    """Function names should be snake_case, not PascalCase."""
    pass


class lowercase_class:
    """Class names should be PascalCase, not snake_case."""

    pass


CONSTANT_value = 42  # Constants should be ALL_CAPS


def normal_function():
    """This is correct snake_case naming."""
    pass
